### DISCUSSION TITLE
DuckDB C API: Arrow functions

### DISCUSSION THREAD

**Discussion by abhiaagarwal** [duckdb/duckdb#12452] [Q&A] [answered]
Hi there,

I'm interested in using the DuckDB C API for some high-performance code, and I'd like to use the Arrow bindings for inoperability with other ecosystems. They're marked as deprecated — is there a recommended migration path to replicate this functionality?

**Comment by Tishj** [comment, marked as answer]
Deprecated in this case does not mean there is an alternative currently.
It's more a heads up that we're not happy with the state of it and will rework it into an alternative method in the future

**Reply by abhiaagarwal** [reply]
I see, thanks! Could a notice be added to the docs to signal this? I read through commit logs and I didn't see this intention expressed anywhere, usually a deprecation implies "this has an alternative".

**Reply by robo-todd** [reply]
I have a very similar use-case/issue. Thanks for this clarification.

It would be extremely helpful if there was some kind of roadmap that at least defined the intention of the project to continue supporting an interface that is similar in capability to the existing arrow-C interface. I would need to access this interface from C or C++ in my applications and reworking code is no problem.

However, I'm concerned that the support for arrow might "go away" which would make selecting DuckDB a fatal dependency mistake in my project. Is there a definitive plan to continue to support "something like this" in the future? I totally understand if the API gets reworked, but we would need to essentially drop DuckDB if it got completely removed somehow.

Any insight on this?
